### PR TITLE
AI Tutor App Restructure

### PR DESCRIPTION
**Pull request by XThomasBU**
Restructuring the codebase, so that we separate the apps and modules, and treat modules as a package. This way we can build multiple apps on the same repo, and keep improving modules independently

### CODE REVIEW

**Comment by XThomasBU** [comment]
Added access restrictions to the bot for emails in https://github.com/DL4DS/dl4ds_tutor/blob/code_restructure/apps/ai_tutor/private/students_encrypted.json, the emails are hashed using a deterministic hash function with a key

File: apps/ai_tutor/private/students_encrypted.json
Comment (trgardos) [inline apps/ai_tutor/private/students_encrypted.json:1]: This file is not private, so maybe doesn't belong in ./private. Perhaps just a readme in that folder to say what is expected. Then maybe students_encrypted.json goes in public?

File: apps/ai_tutor/public/acastusphoton-svgrepo-com.svg
Comment (trgardos) [inline apps/ai_tutor/public/acastusphoton-svgrepo-com.svg:1]: maybe a better folder name for the images would be assets/images?

File: requirements.txt
Comment (trgardos) [inline requirements.txt:1]: should we separate module requirements from app requirements?

**Review by trgardos — COMMENTED** [review]
Left some comments. Curious what you think.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+# Read the contents of requirements.txt
+with open("requirements.txt") as f:
+    requirements = f.read().splitlines()
+
+setup(
+    name="dl4ds_tutor",
+    version="0.1.0",
+    packages=find_packages(),
+    package_dir={"modules": "modules"},
```
Comment (trgardos) [inline setup.py:11]: where does the actual package name come from? This line? Since it shows up in pip as "modules". If so, change to "edubot-core".

**Review by trgardos — COMMENTED** [review]
a few comments on setup.py

File: apps/ai_tutor/private/students_encrypted.json
Comment (XThomasBU) [reply apps/ai_tutor/private/students_encrypted.json:1]: Moved to `public/files`, added initial README

File: apps/ai_tutor/public/acastusphoton-svgrepo-com.svg
Comment (XThomasBU) [reply apps/ai_tutor/public/acastusphoton-svgrepo-com.svg:1]: moved to `public/assets/images/starter_icons`

File: setup.py (hunk 1 above)
Comment (trgardos) [reply setup.py:11]: Sure. we can do that after we transfer the repo.

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+from setuptools import setup, find_packages
+
+# Read the contents of requirements.txt
+with open("requirements.txt") as f:
```
[the remark this answers is not in the document]
Comment (trgardos) [reply setup.py:4]: @xavier -- care to comment on this one?

File: requirements.txt
Comment (trgardos) [reply requirements.txt:1]: @XThomasBU Care to comment?

Comment (XThomasBU) [reply requirements.txt:1]: Okay, can do that in a separate PR, would need to do some testing to figure out which packages are modules specific

Comment (trgardos) [reply requirements.txt:1]: we can make this change after we transfer.

**Review by trgardos — APPROVED** [review]

**Pull request merged**